### PR TITLE
first round of updates

### PR DESCRIPTION
**Pull request by jhamman**
cc @bartnijssen @tbohn @dgergel @yixinmao 

See vic5 slack channel for response to reviewers. 

--- DO NOT MERGE ---

### CODE REVIEW

File: vic5_manuscript.tex [hunk 1]
```diff
@@ -159,7 +159,7 @@ \section{VIC Infrastructure}
...
+  While VIC's original infrastructure may have made coupling within ESMs difficult, it also contributed to the model's wide uptake throughout the hydrologic modeling community. Previous versions of VIC were very portable, required no third-party libraries, were easy to run in single point mode for testing or comparison with point observations, and were trivial to parallelize across many processors and machines by manually subdividing the domain and running each sub-domain on a separate processor. It was this simple infrastructure, combined with the inclusion of built-in convenience tools such as the meteorological variable disaggregator based on the Mountain Climate Simulator (MT-CLIM) algorithms \citep{Thornton_1999,Bohn_2013}, publicly available regional and global forcing and parameter datasets, and publicly available source code, that greatly contributed to VIC's wide adoption in the hydrologic modeling community.
```
Comment (tbohn) [inline vic5_manuscript.tex]: Just to be annoyingly technical, might be worth additionally citing Kimball et al 1997 (or 1998?) for the VP algorithm, for completeness.

File: vic5_manuscript.tex [hunk 2]
```diff
@@ -177,7 +177,7 @@ \section{VIC Infrastructure}
...
+    A practical motivating factor in the development of VIC-5 was to enable coupling of VIC within the Regional Arctic System Model \citep[RASM;][]{Hamman_2016a}, a regional implementation of the Community Earth System Model \citep[CESM;][]{Hurrell_2013}, while maintaining usability as a stand-alone model. In RASM, VIC-5 is directly coupled to CESM's flux coupler \citep[CPL7;][]{Craig_2012} via the Model Coupling Toolkit interface \citep[MCT;][]{Larson_2005}. As implemented within RASM, this interface allows VIC to be coupled to the Weather Research and Forecasting (WRF) atmospheric model \citep{Skamarock_2008} and the RVIC (short for routing-VIC) streamflow routing model \citep{Hamman_2017a} components, while maintaining the legacy and stand-alone model infrastructure, which have a large existing user base. The successful coupling of VIC-5 within RASM is a key demonstration of the benefits of the source code refactor. The new infrastructure will facilitate VIC to be coupled more readily to other modeling systems in the future.
```
Comment (tbohn) [inline vic5_manuscript.tex]: maybe "and the routing-VIC (RVIC)"

File: vic5_manuscript.tex [hunk 3]
```diff
@@ -220,7 +224,7 @@ \section{VIC Infrastructure}
...
 
     \begin{enumerate}
-      \item \textit{Exact restarts}: In the versions of VIC that included MT-CLIM, exact restarts were not possible due to a rolling time average in MT-CLIM. In practice, applications that required exact restarts for short model runs (e.g. hydrologic forecasting) were forced to run the model for long spin-up periods to achieve near-exact restarts.
+      \item \textit{Exact restarts}: In the versions of VIC that included MT-CLIM, exact restarts were not possible due to a rolling time average in MT-CLIM. By exact restarts, we mean that the model was not able to save and restore its state without loosing information. In practice, applications that required the ability to continue previous simulations using the ending model state (e.g. hydrologic forecasting) were forced to run the model for long spin-up periods to achieve satisfactory restart performance (not exact).
```
Comment (tbohn) [inline vic5_manuscript.tex]: 1. should be "losing", not "loosing"
2. how about "In the versions of VIC that included MT-CLIM, exact restarts from a previously-saved state were not possible, due to MT-CLIM's dependence on a rolling 90-day window of previous meteorological forcings.  Because these forcings were not included in the VIC state file, MT-CLIM used a default initial condition during restarts, causing a discontinuity in model state. In practice, applications that required the ability to continue previous simulations using the ending model state (e.g. hydrologic forecasting) were forced to run the model for long spin-up periods to achieve satisfactory (but inexact) restart performance."  I realize this is longer, probably too wordy, but maybe clearer?

File: vic5_manuscript.tex (hunk 1 above)
Comment (bartnijssen) [reply vic5_manuscript.tex]: Nope - people can look at Bohn 2013 or any of the others to get the detailed info. That is not what this paper is about.

File: vic5_manuscript.tex [hunk 4]
```diff
@@ -142,7 +142,7 @@
...
...del to \textit{"play well with others"}. Instead, the model run-time environment focused on simplicity with the goal that VIC could run without the need for specific machine architectures, compilers or third-party libraries. Model development prior to version 5 did not follow a well-defined development track and was directed primarily by the needs of specific model experiments. Relatively little attention was paid to improvements of the model infrastructure, which had remained essentially unchanged since its initial release in the early 1990s, including input/output (I/O) formats. As a consequence, model implementation typically required extensive pre- and post-processing, including reformatting and processing of model input and output files. The few VIC applications within multi-model frameworks \citep[e.g. NASA LIS, ][]{Kumar_2006} required significant custom reconfiguration of the model interface. While VIC model source code was version controlled and publicly available since the early 1990s, model development was not always transparent. The source code repository of record resided on a private machine at the University of Washington, where most of the development was conducted.
```
[the remark this answers is not in the document]
Comment (bartnijssen) [reply vic5_manuscript.tex]: Agreed - I don't think we need to change anything here (note that it is really hard to "spot the 10 differences" in this PR). I have indicated the same (no change) in the reply to reviewers.

File: vic5_manuscript.tex (hunk 2 above)
Comment (bartnijssen) [reply vic5_manuscript.tex]: Nope - RVIC is not an acronym, but simply the name of the new routing model. Since it never was an acronym I don't want to describe it here as one simply to satisfy a reviewer.

File: vic5_manuscript.tex [hunk 5]
```diff
@@ -193,6 +193,10 @@ \section{VIC Infrastructure}
       \item The \textit{Python Driver} provides Python bindings to the functions and data structures of VIC’s physical core. These functions are made available as Python functions using the C Foreign Function Interface for Python (CFFI; \url{http://cffi.readthedocs.io}). Although the \textit{Python Driver} is currently only used for testing, this driver is uniquely suited for future interactive classroom sessions, data assimilation, or large ensemble applications.
     \end{enumerate}
 
+    Additionally, VIC includes a set of general "Shared Driver Utilities" that may be used by the individual drivers. These utilities are ...
+
+    % TODO - finish that thought...
```
Comment (bartnijssen) [inline vic5_manuscript.tex]: Yes - please finish so we can do a final review.

File: vic5_manuscript.tex (hunk 3 above)
Comment (bartnijssen) [reply vic5_manuscript.tex]: in Ted's suggested edit: Way too much detail about MT-CLIM for my taste.

In the original edit, we want to rephrase this. "By exact restarts, we mean that ..." This is a bit clunky, because then it ends up describing what an exact restart is not rather than what it is.

File: vic5_manuscript.tex [hunk 6]
```diff
@@ -252,7 +256,9 @@ \section{VIC Infrastructure}
...
+    The \textit{Build}, \textit{Unit}, \textit{System}, and \textit{Examples} tests are routinely run using the Travis CI continuous integration system (\url{https://about.travis-ci.com/}, \url{https://travis-ci.org/UW-Hydro/VIC}) where the VIC test suite is executed using a multiple of compilers (GNU and Clang), MPI libraries (MPICH and Open-MPI) and computational architectures (Linux vs OS X). The VIC's Travis workflow includes the following steps: 1) selection of virtual machine with preinstalled environments, 2) download of VIC source code and test datasets from GitHub, 3) compile specific driver along with Python unit tests, and 4) run series of tests for specific driver implementation (e.g. image). If any of the tests in these steps fail for any part of the test matrix, Travis returns a failed test status to the VIC GitHub page. This functionality is routinely used in evaluating VIC modifications that are suggested on GitHub via pull requests.
```
Comment (bartnijssen) [inline vic5_manuscript.tex]: "a multiple of compilers" --- change to ---> "multiple compilers"
"Linux vs OS X" --- change to ---> "linux and OS X"
"The VIC's Travis workflow" --- change to ---> "VIC's Travis workflow"
"3) compile" --- change to ---> "compilation of the "
"4) run series of tests for" --- change to ---> "4) execution of test simulations for the "

File: vic5_manuscript.tex [hunk 7]
```diff
@@ -252,7 +256,9 @@ \section{VIC Infrastructure}
...
+    The \textit{Build}, \textit{Unit}, \textit{System}, and \textit{Examples} tests are routinely run using the Travis CI continuous integration system (\url{https://about.travis-ci.com/}, \url{https://travis-ci.org/UW-Hydro/VIC}) where the VIC test suite is executed using a multiple of compilers (GNU and Clang), MPI libraries (MPICH and Open-MPI) and computational architectures (Linux vs OS X). The VIC's Travis workflow includes the following steps: 1) selection of virtual machine with preinstalled environments, 2) download of VIC source code and test datasets from GitHub, 3) compile specific driver along with Python unit tests, and 4) run series of tests for specific driver implementation (e.g. image). If any of the tests in these steps fail for any part of the test matrix, Travis returns a failed test status to the VIC GitHub page. This functionality is routinely used in evaluating VIC modifications that are suggested on GitHub via pull requests.
+
+    % TODO: mention that vic has been tested with PGI, Intel, Cray compilers but they can't be used on travis because of license issues.
```
Comment (bartnijssen) [inline vic5_manuscript.tex]: Go ahead and complete all the TODOs - I can then check out this PR locally and make one more round of edits (and issue a PR against this PR)

File: vic5_manuscript.tex [hunk 8]
```diff
@@ -301,6 +308,10 @@ \section{Results and Discussion}
 
   The development of VIC-5 focused on refactoring the legacy source code to support a range of modern modeling applications. While the release of VIC-5 does not modify existing or add new process representations, it does modernize the supporting infrastructure in the model while preserving legacy functionality. The incorporation of state of the art software engineering tools, such as GitHub and continuous integration, will facilitate the ongoing development and maintenance of the model by the broader VIC developer community.
 
+  % TODO: postulate on how we expect the future of VIC development and applications to progress (reviewer 1).
+
+  % TODO: highlight new applications (e.g. sensitivity analyses, high res modeling) made possible by these infrastructure improvements (reviewer 1).
+
```
[the remark this answers is not in the document]
Comment (bartnijssen) [reply vic5_manuscript.tex]: We can keep this short.

File: vic5_manuscript.tex [hunk 9]
```diff
@@ -319,13 +330,12 @@ \section{VIC Model Source Code and Documentation}
 \subsection{Archive}
 
 \begin{itemize}
-    %  TODO: update to 5.1.0 when release is completed
-	\item \textit{Name}: VIC.5.0.1
-	\item \textit{Persistent identifier}: \url{http://doi.org/10.5281/zenodo.267178}
+	\item \textit{Name}: VIC.5.1.0
```
Comment (bartnijssen) [inline vic5_manuscript.tex]: revert to 5.0.1 - I don't want to hold this up for 5.1.0

File: vic5_manuscript.tex (hunk 3 above)
Comment (dgergel) [reply vic5_manuscript.tex]: How about something like: "In the versions of VIC that included MT-CLIM, exact restarts were not possible due to a rolling time average. By removing MT-CLIM in VIC.5, it is now possible to end a run and save the model state at that timestep and then start a new model run with exactly the same model state. This improvement has widespread applicability in hydrologic modeling and is particularly important for forecasting applications."

File: vic5_manuscript.tex (hunk 6 above)
Comment (dgergel) [reply vic5_manuscript.tex]: "This functionality is routinely used in evaluating VIC modifications that are suggested on GitHub via pull requests." ---> "This functionality is routinely used in evaluating VIC source code modifications and feature developments that are made on GitHub via pull requests.

File: vic5_manuscript.tex [hunk 10]
```diff
@@ -262,7 +268,7 @@ \section{VIC Infrastructure}
...
+    As part of the development of VIC-5, we have implemented an updated, comprehensive documentation website for VIC (\url{http://vic.readthedocs.io}). While previous versions of the VIC model did have a documentation website, edits were only possible by the VIC administrator, and the documentation was maintained separately from the model source code. In VIC-5, the documentation content is part of the VIC source code repository and is hosted on ReadTheDocs, a public documentation hosting service. A key benefit of the updated documentation is the ability to provide versioned documentation that matches specific model releases. This website is a particularly valuable tool for new users as it combines tutorials for how to setup and run VIC with a detailed description of the model itself.
```
Comment (dgergel) [inline vic5_manuscript.tex]: It might be useful to add a citation for the read the docs hosting service

File: vic5_manuscript.tex [hunk 11]
```diff
@@ -286,7 +293,7 @@ \section{Results and Discussion}
...
...r coupling intervals (e.g. 20 minutes) than the timesteps that have traditionally been used in VIC (hourly to daily). Fig.~\ref{fig:vic_scaling} demonstrates the scaling performance for the two model configurations. For the simpler model configuration (A), VIC run with MPI shows high throughput, but extending parallelization beyond four to eight nodes (144 to 288 cores) provides little speedup. This behavior is indicative of a program that is I/O limited. For the more complex model configuration (B), VIC has significantly lower throughput, but scales efficiently beyond 32 nodes (1,152 cores). In this case, VIC spends much more time inside the physics code than in the I/O code of the model. For example, a comparison of the timing logs for two simulations that each use 144 MPI tasks shows that configuration A spends 48\% on I/O operations, while configuration B spends only 28\%. In practice, the scaling performance for most parallel VIC applications is predominantly a function of a computer's interconnect and disk speeds. We expect that VIC's scaling performance would be therefore improved on machines with more performant file systems (e.g. solid-state-disks) or faster interconnects.
```
Comment (dgergel) [inline vic5_manuscript.tex]: "In this case, VIC spends much more time inside the physics code than in the I/O code of the model." ---> "In this case, the physics code is much more time intensive than the I/O code of the model."

File: vic5_manuscript.tex [hunk 12]
```diff
@@ -439,10 +449,12 @@ \subsection{Versioned Documentation}
 \clearpage
 \begin{figure*}[t]
 \includegraphics[width=12cm]{VIC_science_tests_fluxes.png}
-\caption{Example summary output from the VIC-5 science testing package comparing the annual cycle of latent heat (top) and sensible heat (bottom) for four sample locations for VIC-5 (blue) to FluxNet observations (red).}
+\caption{Example summary output from the VIC-5 science testing package comparing the annual cycle of latent heat (top) and sensible heat (bottom) for four sample locations for VIC-5 (blue) to FluxNet observations (red). The shaded range represents the interanual variability (+/- 0.5 std. dev.).}
```
Comment (dgergel) [inline vic5_manuscript.tex]: interanual ---> interannual

File: vic5_manuscript.tex [hunk 13]
```diff
@@ -551,7 +563,7 @@ \subsection{Versioned Documentation}
 \clearpage
 \begin{table}[]
   \centering
-  \caption{Summary of major VIC developments present in VIC-5, focusing on improved process representation, and the model versions they were added in since the original publication of \citep{Liang_1994}.}
+  \caption{Summary of major VIC developments present in VIC-5, focusing on improved process representation, and the model versions that were added in since the original publication of \citet{Liang_1994}.}
```
Comment (dgergel) [inline vic5_manuscript.tex]: Shouldn't "representation" be "representations"

File: vic5_manuscript.tex (hunk 4 above)
[the remark this answers is not in the document]
Comment (jhamman) [reply vic5_manuscript.tex]: okay, I've reverted this change.

File: vic5_manuscript.tex (hunk 2 above)
Comment (jhamman) [reply vic5_manuscript.tex]: I've removed the blip: `(short for routing-VIC)`

File: vic5_manuscript.tex (hunk 5 above)
Comment (jhamman) [reply vic5_manuscript.tex]: done.

File: vic5_manuscript.tex (hunk 8 above)
[the remark this answers is not in the document]
Comment (jhamman) [reply vic5_manuscript.tex]: This is the last text edit I need to make. Will take me a few minutes to think through this point...

File: vic5_manuscript.tex (hunk 1 above)
Comment (tbohn) [reply vic5_manuscript.tex]: OK, but same can be said for Thornton and Running 1999, which is cited.  Perhaps Thornton and Running should be omitted.

**Comment by dgergel** [comment]
Full read-through comments: 

- Line 143: UW Drought Monitoring site link does not work (the hydro.washington.edu sites are no longer active I believe)
- Line 145: `custom reconfiguration` --> `customized reconfiguration`
- Line 151: `VIC's useful life` --> `the usefulness of VIC`
- Line 160: typo in `common` when writing out what NetCDF stands for
- Line 208 under Parallel Computing: `often times` --> `oftentimes`
- Line 210: `default domain composition` should be `default domain decomposition`
- Line 218, under Input and Output: I think `controlled by compile time option` should be `controlled by compile-time option`
- Line 232: `transparency, flexibility, and accessibility of the model` - the Oxford comma! it should be `transparency, flexibility and accessibility of the model`
- Line 257: typo in last sentence `VIC-5 has also ben tested with the PGI, Intel, and Cray compilers but they are not used in Travis CI because of licensing issues.`, `ben` --> `been`
- Line 261: I would say `collaborative coding` instead of `social coding`, and remove comma after `line-by-line code reviews`
- Line 267: add parentheses around the link for Read the Docs
- Fig 5 caption: should be `MPI-only` I think 
- Line 307: add comma after `combined with a suite of modern software tools`, add `the` before `development process`
- Line 312 under Code Availability: remove comma after `demonstrations for this paper` 
- Line 319: add `of` before the citation of Liang et al 1994 and make `process representation` plural, also `citations for the relevant literature` instead of `to`
- Author contributions: add `of` before `the source code configuration`
- Table 1: under climate change impact analysis could cite Gergel et al 2017, since the trademark scientist comment is to suggest citing their own work 

cc @jhamman @bartnijssen @yixinmao @tbohn

**Comment by nemequ** [comment]
> Line 257: typo in last sentence VIC-5 has also ben tested with the PGI, Intel, and Cray compilers but they are not used in Travis CI because of licensing issues., ben --> been

PGI has a community edition which [can be used on Travis](https://github.com/nemequ/pgi-travis).

**Comment by jhamman** [comment]
Thank you @nemequ. I will update the manuscript accordingly.
